### PR TITLE
Introduce WgKey type

### PR DESCRIPTION
**Pull request by donkey-hotei**

### CODE REVIEW

File: althea_kernel_interface/src/create_wg_key.rs [hunk 1]
```diff
@@ -45,13 +47,14 @@ impl KernelInterface {
...
         let mut pubkey_str = String::from_utf8(output.stdout)?;
+
         privkey_str.truncate(44);
         pubkey_str.truncate(44);
 
-        Ok(WgKeypair {
-            public: pubkey_str,
-            private: privkey_str,
-        })
+        let private = WgKey::from_str(&privkey_str).unwrap();
```
Comment (jkilpatr) [inline althea_kernel_interface/src/create_wg_key.rs]: why use unwraps rather than ?

Comment (donkey-hotei) [reply althea_kernel_interface/src/create_wg_key.rs]: No real reason!

**Comment by donkey-hotei** [comment]
Yeah, I haven't finished integrating this across the repo as of yet. There could probably be more thorough tests for validating on malformed keys. I wasn't sure why, before, the `validate_wg_key` function checked that the keys were _exactly_ 44 characters when I couldn't find a specification for it in WireGuard docs.

**Comment by donkey-hotei** [comment]
Yeah, I haven't finished integrating this across the repo as of yet. There could probably be more thorough tests for validating on malformed keys. I wasn't sure why, before, the `validate_wg_key` function checked that the keys were _exactly_ 44 characters when I couldn't find a specification for it in WireGuard docs.

**Comment by donkey-hotei** [comment]
@jkilpatr Should I close this since you've opened up a new PR #357 ?

**Comment by donkey-hotei** [comment]
So, I'm going to keep this one - more up-to-date and without merge conflicts.

**Comment by donkey-hotei** [comment]
CI build failing with:
```

**Comment by donkey-hotei** [comment]
Wanted to reference the related issue here for posterity, #337 .

**Pull request closed without merging**